### ISSUE TITLE
Plotting from geocode consistently errors

### ISSUE THREAD

**Issue by scottishbee** [gmplot/gmplot#82]
Running the example code 
`gmplot.GoogleMapPlotter.from_geocode("San Francisco")`
used to work exactly as described.  

Now it consistently errors `list index out of range`
Swapping for lat/lon version works fine (`gmplot.GoogleMapPlotter(37.766956, -122.438481, 13)`)

Full error

> ---------------------------------------------------------------------------
> IndexError                                Traceback (most recent call last)
> <ipython-input-130-02f152c69777> in <module>()
> ----> 1 a_map = gmplot.GoogleMapPlotter.from_geocode("San Francisco")
> 
> /Users/scottieb/anaconda/envs/GDAL/lib/python3.6/site-packages/gmplot/gmplot.py in from_geocode(cls, location_string, zoom)
>      32 
>      33     @classmethod
> ---> 34     def from_geocode(cls, location_string, zoom=13):
>      35         lat, lng = cls.geocode(location_string)
>      36         return cls(lat, lng, zoom)
> 
> /Users/scottieb/anaconda/envs/GDAL/lib/python3.6/site-packages/gmplot/gmplot.py in geocode(self, location_string)
>      40         geocode = requests.get(
>      41             'http://maps.googleapis.com/maps/api/geocode/json?address="%s"' % location_string)
> ---> 42         geocode = json.loads(geocode.text)
>      43         latlng_dict = geocode['results'][0]['geometry']['location']
>      44         return latlng_dict['lat'], latlng_dict['lng']
> 
> IndexError: list index out of range

**Comment by vgm64** [comment]
That has been an opaque failure mode for a long time. gmplot should really expose what geocode.text in the error message for better diagnosing.

**Comment by DakotaHoosier** [comment]
Two problems with GoogleMapPlotter.from_geocode

- The API key isn't passed and it is required.
- The request needs to use https:// instead of http://

This can be fixed by using HTTPS instead of HTTP (you'lll need a Google Maps API key to see the error with http vs. https):

This works:
**https**://maps.googleapis.com/maps/api/geocode/json?address=%22%ssan%20francisco%22&&key=mygoogleapikey

This returns an error:
**http**://maps.googleapis.com/maps/api/geocode/json?address=%22%ssan%20francisco%22&&key=mygoogleapikey
```
{
   "error_message" : "Requests to this API must be over SSL. Load the API with \"https://\" instead of \"http://\".",
   "results" : [],
   "status" : "REQUEST_DENIED"
}
```

**Comment by vgm64** [comment]
I encountered this recently when using the package after a long lapse. I'll try to make some improvements here in the next month.

**Comment by sandrotosi** [comment]
the fix is at : https://github.com/vgm64/gmplot/issues/58#issuecomment-430849353

**Comment by DakotaHoosier** [comment]
Yes, I'd like to see this. Fix is needed and docs need an example that shows API key. It's a reality that Google expects this now.

**Comment by imjackicle** [comment]
> the fix is at : [#58 (comment)](https://github.com/vgm64/gmplot/issues/58#issuecomment-430849353)

It worked for me. Thank you !

**Comment by frslm** [comment]
Fixed in #111.
